scripts/help-utils.js

```diff
@@ -0,0 +1,217 @@
+});
+
+const freshdeskLocale = locale => {
+    // map between Transifex locale and Freshdesk. Two letter codes are usually fine
+    const localeMap = {
+        'es_419': 'es-LA',
+        'ja': 'ja-JP',
+        'ja-Hira': 'ja-JP',
+        'lv': 'lv-LV',
+        'nb': 'nb-NO',
+        'nn': 'nb-NO',
+        'pt': 'pt-PT',
+        'pt_BR': 'pt-BR',
+        'ru': 'ru-RU',
+        'sv': 'sv-SE',
+        'zh_CN': 'zh-CN',
+        'zh_TW': 'zh-TW'
+    };
+    return localeMap[locale] || locale;
+};
+
+// Promisify Transifex async/callbacks to make them easier to use`
+const getResources = util.promisify(TX.resourcesSetMethod.bind(TX));
+const getResourceInfo = util.promisify(TX.resourcesInstanceMethods.bind(TX));
+const getTranslation = util.promisify(TX.translationInstanceMethod.bind(TX));
+
+/**
+ * Pull metadata from Transifex for the scratch-help project
+ * @return {Promise} results array containing:
+ *                      languages: array of supported languages
+ *                      folders: array of tx resources corrsponding to Freshdesk folders
+ *                      names: array of tx resources corresponding to the Freshdesk metadata
+ */
+exports.getInputs = async () => {
+    const resources = await getResources(TX_PROJECT);
+    // there are three types of resources differentiated by the file type
+    const folders = resources.filter(resource => resource.i18n_type === 'STRUCTURED_JSON');
+    const names = resources.filter(resource => resource.i18n_type === 'KEYVALUEJSON');
+    // ignore the yaml type because it's not possible to update via API
+
+    // Lookup available languages by getting metadata for a resource, they all have the
+    // same set of languages, so it doesn't matter which one you get
+    const resourceInfo = await getResourceInfo(TX_PROJECT, resources[0].slug, true);
+    const languages = resourceInfo.available_languages.map(l => l.code);
+
+    return Promise.all([languages, folders, names]); // eslint-disable-line no-undef
+};
+
+/**
+ * internal function to serialize saving category and folder name translations to avoid Freshdesk rate limit
+ * @param  {[type]}  json     [description]
+ * @param  {[type]}  resource [description]
+ * @param  {[type]}  locale   [description]
+ * @return {Promise}          [description]
+ */
+const serializeNameSave = async (json, resource, locale) => {
+    for (let [key, value] of Object.entries(json)) {
+        // key is of the form <name>_<id>
+        const words = key.split('_');
+        const id = words[words.length - 1];
+        let status = 0;
+        if (resource.name === 'categoryNames_json') {
+            status = await FD.updateCategoryTranslation(id, freshdeskLocale(locale), {name: value});
+        }
+        if (resource.name === 'folderNames_json') {
+            status = await FD.updateFolderTranslation(id, freshdeskLocale(locale), {name: value});
+        }
+        if (status === -1) {
+            process.exitCode = 1;
```

Comment:
This style seems fine for now but these tools would be more reusable in the future if they didn't assume they were running in a script, and either returned the status value, or I think better would be to return a result in case of success, and throw `Error`s otherwise.